### PR TITLE
fix: Incorrect Renamed event args on unix

### PR DESCRIPTION
**Pull request by pw-sgr**
#899 [Regression](https://github.com/Testably/Testably.Abstractions/issues/899#issuecomment-3711951580)

### CODE REVIEW

**Comment by pw-sgr** [comment]
@vbreuss implemented the correct behaviors, tho one problem I have with windows are the `Changed` events, which fire almost randomly and not repreducable between Windows Server 2025 (github image) and my local Windows 11 Pro 24H2 26100.7462, so for now I ignored them

File: Source/Testably.Abstractions.Testing/FileSystem/FileSystemWatcherMock.cs [hunk 1]
```diff
@@ -736,6 +956,60 @@ public WaitForChangedResultMock(
 		public bool TimedOut { get; }
 	}
 
+	[StructLayout(LayoutKind.Auto)]
+	private readonly struct RenamedContext(
+		bool comesFromOutside,
+		bool comesFromInside,
+		bool goesToInside,
+		int oldSubDirectoryCount
+	) : IEquatable<RenamedContext>
```
Comment (vbreuss) [inline Source/Testably.Abstractions.Testing/FileSystem/FileSystemWatcherMock.cs]: Why did you implement `IEquatable<T>`? You use the `RenamedContext` as a DTO without comparing it anywhere.

File: Tests/Testably.Abstractions.Tests/FileSystem/FileSystemWatcher/MoveTests.cs [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+					                x.OldFullPath, insideTarget, StringComparison.Ordinal
+				                )
+				);
+		}
+	}
+
+	private static bool EqualsOrdinal(string? x, string? y)
+	{
+		return string.Equals(x, y, StringComparison.Ordinal);
+	}
```
Comment (vbreuss) [inline Tests/Testably.Abstractions.Tests/FileSystem/FileSystemWatcher/MoveTests.cs]: Do we need this method, when the `Satisfies`-expectations were re-written?

**Review by vbreuss — COMMENTED** [review]
Wow, I didn't realize how many edge-cases we could/should consider. I'm impressed that you got it working (I already gave up on it 👍).

File: Source/Testably.Abstractions.Testing/FileSystem/FileSystemWatcherMock.cs [hunk 3]
```diff
@@ -532,39 +535,260 @@ private void Stop()
...
 			{
 				Renamed?.Invoke(this, eventArgs);
 			}
 		}
 	}
 
+	private void TriggerMacRenameNotification(ChangeDescription item, RenamedContext context)
+	{
+		// Premise
+		// context.ComesFromOutside == true && context.GoesToInside == true covered
```
[the remark this answers is not in the document]
Comment (pw-sgr) [reply Source/Testably.Abstractions.Testing/FileSystem/FileSystemWatcherMock.cs]: Sounds good, do you prefer `Debug.Assert` or an exception? And if exception, should this also be in the ExceptionFactory, I'm unsure since they aren't for the file system, but for the maintainer

File: Tests/Testably.Abstractions.Tests/FileSystem/FileSystemWatcher/MoveTests.cs [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+			await That(deletedEvent!).Satisfies(x => x.ChangeType == WatcherChangeTypes.Deleted
+			                                         && string.Equals(
+				                                         x.Name, expectedDeletedName,
+				                                         StringComparison.Ordinal
+			                                         )
+			                                         && string.Equals(
+				                                         x.FullPath, insideTarget,
+				                                         StringComparison.Ordinal
+			                                         )
+			);
```
[the remark this answers is not in the document]
Comment (pw-sgr) [reply Tests/Testably.Abstractions.Tests/FileSystem/FileSystemWatcher/MoveTests.cs]: Changed

File: Source/Testably.Abstractions.Testing/FileSystem/FileSystemWatcherMock.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (vbreuss) [reply Source/Testably.Abstractions.Testing/FileSystem/FileSystemWatcherMock.cs]: I think this gets out of hand quickly. If it gets this complicated, it is probably better to leave it as is. The change is already quite complex to comprehend. Maybe we can consider a refactoring later on.

[the remark this answers is not in the document]
Comment (pw-sgr) [reply Source/Testably.Abstractions.Testing/FileSystem/FileSystemWatcherMock.cs]: Ok, left the channel as is and replaced the comments with a method with `Debug.Assert`
